android/java/org/chromium/chrome/browser/crypto_wallet/util/BalanceHelper.java

```diff
@@ -249,37 +249,35 @@ private static void processP3ACoinNetworks(@CoinType.EnumType int coinType,
-                                    getNativeAssetsBalances(jsonRpcService, network,
-                                            accountInfoArray, getNativeAssetsBalancesContext);
-                                    nativeAssetsBalancesResponses.add(
-                                            getNativeAssetsBalancesContext);
+        keyringService.getAllAccounts(allAccounts -> {
+            for (NetworkInfo network : networks) {
+                AccountInfo[] accountInfoArray = AssetUtils.filterAccountsByNetwork(
+                        allAccounts.accounts, network.coin, network.chainId);
 
-                                    // Tokens balances.
-                                    GetBlockchainTokensBalancesResponseContext
-                                            getBlockchainTokensBalancesContext =
-                                                    new GetBlockchainTokensBalancesResponseContext(
-                                                            multiResponse.singleResponseComplete);
-                                    getBlockchainTokensBalances(jsonRpcService, network,
-                                            accountInfoArray, tokens,
-                                            getBlockchainTokensBalancesContext);
-                                    blockchainTokensBalancesResponses.add(
-                                            getBlockchainTokensBalancesContext);
-                                });
-                    });
-        }
+                TokenUtils.getUserAssetsFiltered(
+                        braveWalletService, network, coinType, TokenUtils.TokenType.ALL, tokens -> {
+                            // Assets balances.
+                            GetNativeAssetsBalancesResponseContext getNativeAssetsBalancesContext =
+                                    new GetNativeAssetsBalancesResponseContext(
+                                            multiResponse.singleResponseComplete);
+                            getNativeAssetsBalances(jsonRpcService, network, accountInfoArray,
+                                    getNativeAssetsBalancesContext);
+                            nativeAssetsBalancesResponses.add(getNativeAssetsBalancesContext);
+
+                            // Tokens balances.
+                            GetBlockchainTokensBalancesResponseContext
+                                    getBlockchainTokensBalancesContext =
+                                            new GetBlockchainTokensBalancesResponseContext(
+                                                    multiResponse.singleResponseComplete);
+                            getBlockchainTokensBalances(jsonRpcService, network, accountInfoArray,
+                                    tokens, getBlockchainTokensBalancesContext);
+                            blockchainTokensBalancesResponses.add(
+                                    getBlockchainTokensBalancesContext);
+                        });
+            }
+        });
```

Comment:
just moved keyringService.getAllAccounts as the first call so for each network can filter accounts for all accounts.